fireant/slicer/transformers/datatables.py

```diff
@@ -24,17 +25,30 @@ def _safe(value):
         # Cannot transform np.int64 to json
         return int(value)
 
+    if isinstance(value, np.float64):
+        return float(value)
+
     return value
 
 
 def _pretty(value, schema):
     precision = schema.get('precision')
 
-    if value and isinstance(value, float) and precision:
-        value = round(value, precision)
+    if isinstance(value, float):
+        if precision is not None:
+            float_format = '%d' if precision == 0 else '%.{}f'.format(precision)
+            value = lc.format(float_format, value, grouping=True)
+        elif value.is_integer():
+            float_format = '%d'
+            value = lc.format(float_format, value, grouping=True)
```

Comment:
it'd be nice if this stuff would happen more implicitly. Unfortunately iterating through rows in pandas turns everything into a floating point.